### PR TITLE
layers: Clarify Descriptor Buffer Binding limits

### PR DESCRIPTION
**Pull request by spencer-lunarg**
closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/8835

### CODE REVIEW

File: tests/unit/descriptor_buffer.cpp [hunk 1]
```diff
@@ -1668,3 +1669,58 @@ TEST_F(NegativeDescriptorBuffer, TexelBufferFormat) {
...
+        GTEST_SKIP() << "maxResourceDescriptorBufferBindings  is not 1";
+    }
+
+    VkDescriptorSetLayoutBinding binding = {0, VK_DESCRIPTOR_TYPE_STORAGE_BUFFER, 1, VK_SHADER_STAGE_ALL, nullptr};
+    vkt::DescriptorSetLayout ds_layout(*m_device, binding, VK_DESCRIPTOR_SET_LAYOUT_CREATE_DESCRIPTOR_BUFFER_BIT_EXT);
+
+    VkDeviceSize ds_layout_size = 0;
+    vk::GetDescriptorSetLayoutSizeEXT(device(), ds_layout.handle(), &ds_layout_size);
+
+    ds_layout_size = Align(ds_layout_size, descriptor_buffer_properties.descriptorBufferOffsetAlignment);
```
Comment (artem-lunarg) [inline tests/unit/descriptor_buffer.cpp:1689]: `descriptorBufferOffsetAlignment` should be used to ensure that buffer address  `VkDescriptorBufferBindingInfoEXT::address` (line 1713) is aligned properly. Also can't find that spec guarantees that buffer address that corresponds to the beginning of the buffer has correct alignment. This is used in other tests, can be fixed as a separate cleanup.

When I experimented with this I assumed that mapped buffer pointer satisfies this requirement (based on `VK_BUFFER_USAGE_RESOURCE_DESCRIPTOR_BUFFER_BIT_EXT` flag), but that's probably not a robust solution: https://github.com/kennyalive/vulkan-base/blob/f1f9fb2562d2eecb2eae8215f9e02ef82c27bbc3/src/demo.cpp#L187

**Review by artem-lunarg — APPROVED** [review]
LGTM. One comment how alignment requirement is treated by the descriptor buffer tests.

File: tests/unit/descriptor_buffer.cpp (hunk 1 above)
Comment (artem-lunarg) [reply tests/unit/descriptor_buffer.cpp:1689]: I would be happy if spec guarantees some alignments of buffer address of the beginning of the buffer. Let me know if you can find something. If no such thing, it actually feels like a spec design issue.

Comment (spencer-lunarg) [reply tests/unit/descriptor_buffer.cpp:1689]: will take a look, this is a copy-and-paste test so need to fix everywhere

**Pull request merged**